### ISSUE TITLE
Auto-Playlist

### ISSUE THREAD

**Issue by joesch2** [mixxxdj/mixxx#13403]
### Feature Description

It would be helpful if the system could create dynamic playlists based on song tags. The type of dynamic lists should be configurable.

Example: 
I enable dynamic playlists based on tag "genre". The system displays a folder "Genre” in the archive tree under playlists, in which there is a folder for each genre found that contains the titles that contain the genre as a tag. 

It would also be nice to have an entry “Auto Playlist” in the root directory, under which all automatic playlists-folder are displayed.

**Comment by daschuer** [comment]
Mixxx can already populate the Auto DJ playlist from crates. So you may create crates with the desired genres and make them an Auto DJ track source. 

Sure the used magic formula can be improved all help is welcome.

**Comment by joesch2** [comment]
Thanks for the feedback!

In my use case, tracks newly added to the collection should be automatically added to the corresponding record box at the same time based on the Genre tag. For me it's not about using it with the AutoDJ.

Example: Example: There is a “Walz” and a “Jive” record box. I'm adding three tracks to the collection with "Jive" as the genre tag. The titles are not only visible in the track collection (Tracks folder) but are also automatically visible in the “Jive” box.
This is helpful, for example, in dance sports training when a specific dance needs to be practiced.

**Comment by joesch2** [comment]
Maybe it would be possible to evaluate the name of the record box for this function. If this consists of a valid search term, the box will be filled dynamically.
![grafik](https://github.com/mixxxdj/mixxx/assets/18378305/6f124bde-c4ec-4934-8140-ae32a92b8278)

**Comment by Eve00000** [comment]
I think joesch2 wants something like "smart playlists" in iTunes: playlist that are defined as a result of a query that keeps on being updated.
= kind of pinning the result of a query to a new crate.
This way you could have crates  Springsteen (artist like 'Springsteen'), fast (title like 'fast'), Mozart (composer like 'Mozart'), Jive (genre like 'jive').
In iTunes it's like the more smart playlists there are, the slower iTunes gets. Therefore they made an option 'keep updated" that means you can deactivate the continuous checking and trigger the playlist-update manually.
It would be a nice feature. In iTunes I use this to correct tags, or to list the tracks added in the last month, 100 most played, classical tracks, tracks without artwork...
Executing a query is much faster in Mixxx so it's not hard to put the result of a query in a new/ existing crate (execute query -> select tracks - > right click -> crates -> select the crate or choose new crate). 
Next step could be automatic coloring of the tracks with the dedicated crate color:-)

**Comment by ronso0** [comment]
Is this a dupe of #5575 (smart crates)?

Currently, something similiar (without a nice tree interface) can be accomplished with saved search queries.

**Comment by Eve00000** [comment]
Yes, I think it is, you've got a good memory and knowledge of all issues/requests ....

**Comment by ronso0** [comment]
Well, that topic popped up a few times during the library redesign proposal and when I worked on saved queries.

Shall we close this one and continue the discussion in #5575?
(which I just renamed to Smart crates / save and recall search queries")

**Comment by daschuer** [comment]
In addition this one has the aspects of new tracks. 

We have on one hand Smart Crates, that are from my understanding repeated auto populated and on the other hand an action for new discovered tracks. 
I can imagine for instance an "inbox" crate where I can move tracks from.

**Comment by Eve00000** [comment]
This 'inbox' can be a smart crate with 'Date added - diff < 1 day'?

**Comment by ronso0** [comment]
If we stick to the idea that Smart Crates are simpy search queries, new tracks are "added" on each crate select.

I like the Inbox idea, but it's a bit different though.
When filtering by date added (like currently done in Analyze -> [ ] New) we don't know which new tracks have been prepared / listened / sorted into crates.
I think we could have a **New** item, as child of **Tracks**, which would be a special crate where we add new tracks discovered during a library scan. That list can be cleaned up automatically (remove track as soon as it's moved to a crate) or manually, like in email inboxes.

This is different from Smart Crates, so let's move that idea to a new feature request.

And let's decide whether to continue with Smart Crates here or in #5575 (by now I think we should stay here).

**Comment by ronso0** [comment]
related: #13192

**Comment by Eve00000** [comment]
I would like to invite you all to test PR #14182  "Smarties (or SmartList or SmartCrate or QueryCrate or QueryList or ...) open for feedback."
I 'm realy keen on feedback.

**Comment by joesch2** [comment]
> I would like to invite you all to test PR [#14182](https://github.com/mixxxdj/mixxx/pull/14182) "Smarties (or SmartList or SmartCrate or QueryCrate or QueryList or ...) open for feedback." I 'm realy keen on feedback.

Sorry for my question - but testing on github it's new for me ;-) How can I deploy it? Can a test version be installed on top of the “production” version without side effects?

**Comment by Eve00000** [comment]
> Sorry for my question - but testing on github it's new for me ;-) How can I deploy it? Can a test version be installed on top of the “production” version without side effects?

Dont feel sorry, here's an explanation:
- You can build Mixxx yourself, therefore you need to follow the nice explained manual in the wiki: [Compile Mixxx From Source Code](Compile Mixxx From Source Code)
- You can download and installed builds that are created during the controls and compilation in github. In the PR (pull request) under 'checks'  scroll down and you can find special Mixxx-builds containing all 'merged' updates to that moment + the functions of the PR.

As smarties hads the label 'draft' at the moment, you need to go to my own repo (repository) to find the builds.
You can find them [here](https://github.com/Eve00000/mixxx/pull/13/checks)

Concerning installing different versions, always safeguard your data, you can find a wiki-article [here](https://github.com/mixxxdj/mixxx/wiki/Safeguard%20your%20Digital%20DJ%20Data). If you saveguard your data (database, settings, controllermapping) you can install another version (more versions) to test.
As the smarties-pr makes changes in the database (add smarties-table) you MUST make a copy first thn you can experiment as much as you want and afterwards de-install the testversion and restore you personal data to the original location.

This way you can test as much as you want.
If you want to test more pr's at the same time, you need to build Mixxx yourself with all the pr's 'merged' into your personal test-version.

Have fun!

**Comment by joesch2** [comment]
Unfortunately I didn't manage to build a Windows version. Sorry... But I think the functionality shown in the "preview" (https://github.com/mixxxdj/mixxx/pull/14182) works well. Thanks!

**Comment by Eve00000** [comment]
@joesch2, that's a pitty, where did it go wrong?
You could also download the installer from the pr.
I'm glad you like it.
